source/MaterialXRender/Mesh.cpp

```diff
@@ -42,8 +43,17 @@ Mesh::Mesh(const string& name) :
 
 MeshStreamPtr Mesh::generateNormals(MeshStreamPtr positionStream)
 {
+    const string normalStreamName = "i_" + MeshStream::NORMAL_ATTRIBUTE;
+
+    // Return if already exists
+    MeshStreamPtr normalStream = getStream(normalStreamName);
+    if (normalStream)
+    {
+        return normalStream;
+    }
```

Comment:
Following up on the discussion above, wouldn't we want to omit these existence checks from the `generateNormals`, `generateTextureCoordinates`, and related helper methods?  Ideally we don't want this check to be performed twice, and the use of "generate" in the names of these methods suggests that it's the caller that should be responsible for any caching logic.

Reply:
I've made the change to perform existence test outside now.